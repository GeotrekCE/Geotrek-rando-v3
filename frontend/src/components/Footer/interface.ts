export interface SocialNetwork {
  id: string;
  url: string;
}

export interface PortalContact {
  name: string;
  addressLine1: string;
  addressLine2: string;
  number: string;
  mail: string;
}

export interface PortalLinkStatic {
  label: string;
  url: string;
}

interface PortalLinkDynamic {
  informationID: number;
}

export type PortalLink = PortalLinkStatic | PortalLinkDynamic;

export interface FooterConfigInput {
  socialNetworks?: SocialNetwork[];
  links?: PortalLink[];
  contact?: Partial<PortalContact>;
  footerTopHtml: {
    [key: string]: undefined | string;
    default: undefined | string;
  };
  footerBottomHtml: {
    [key: string]: undefined | string;
    default: undefined | string;
  };
}

export interface FooterConfigOutput {
  socialNetworks?: SocialNetwork[];
  links?: PortalLinkStatic[];
  contact?: Partial<PortalContact>;
  footerTopHtml: {
    [key: string]: undefined | string;
    default: undefined | string;
  };
  footerBottomHtml: {
    [key: string]: undefined | string;
    default: undefined | string;
  };
}
