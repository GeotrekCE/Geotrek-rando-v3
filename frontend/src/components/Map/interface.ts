import { TileLayerOptions } from 'leaflet';
export interface TileLayer {
  url: string;
  options?: TileLayerOptions;
  bounds?: string;
}

export interface MapConfig {
  searchMapCenter: number[];
  maximumZoomLevel: number;
  searchMapZoom: number;
  mapClassicLayers: TileLayer[];
  mapSatelliteLayers: TileLayer[] | null;
  mapOfflineLayer: TileLayer;
  zoomAvailableOffline?: number[];
}
