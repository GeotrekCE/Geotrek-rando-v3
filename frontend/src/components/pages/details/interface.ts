export type DetailsSectionTrekNames =
  | 'presentation'
  | 'medias'
  | 'itinerancySteps'
  | 'poi'
  | 'description'
  | 'forecastWidget'
  | 'altimetricProfile'
  | 'sensitiveAreas'
  | 'practicalInformations'
  | 'accessibility'
  | 'more'
  | 'source'
  | 'report'
  | 'touristicContent'
  | 'reservationWidget';

export type DetailsSectionTouristicContentNames =
  | 'presentation'
  | 'practicalInformations'
  | 'accessibility'
  | 'contact'
  | 'forecastWidget'
  | 'source';

export type DetailsSectionTouristicEventNames =
  | 'presentation'
  | 'description'
  | 'forecastWidget'
  | 'practicalInformations'
  | 'source'
  | 'touristicContent';

export type DetailsSectionOutdoorSiteNames =
  | 'presentation'
  | 'medias'
  | 'poi'
  | 'description'
  | 'subsites'
  | 'courses'
  | 'sensitiveAreas'
  | 'practicalInformations'
  | 'access'
  | 'forecastWidget'
  | 'source'
  | 'more'
  | 'touristicContent';

export type DetailsSectionOutdoorCourseNames =
  | 'presentation'
  | 'description'
  | 'gear'
  | 'equipment'
  | 'poi'
  | 'sensitiveAreas'
  | 'practicalInformations'
  | 'touristicContent'
  | 'forecastWidget';

interface SectionsProps {
  display: boolean;
  anchor: boolean;
  order: number;
  template?: string;
}

export type SectionsTrek = SectionsProps & {
  name: DetailsSectionTrekNames;
};

export type SectionsTouristicContent = SectionsProps & {
  name: DetailsSectionTouristicContentNames;
};

export type SectionsTouristicEvent = SectionsProps & {
  name: DetailsSectionTouristicEventNames;
};

export type SectionsOutdoorSite = SectionsProps & {
  name: DetailsSectionOutdoorSiteNames;
};

export type SectionsOutdoorCourse = SectionsProps & {
  name: DetailsSectionOutdoorCourseNames;
};

export type Sections = {
  trek: SectionsTrek[];
  touristicContent: SectionsTouristicContent[];
  touristicEvent: SectionsTouristicEvent[];
  outdoorSite: SectionsOutdoorSite[];
  outdoorCourse: SectionsOutdoorCourse[];
};

export type SectionsTypes =
  | SectionsTrek
  | SectionsTouristicContent
  | SectionsTouristicEvent
  | SectionsOutdoorSite
  | SectionsOutdoorCourse;

export type DetailsConfig = {
  sections: Sections;
};
