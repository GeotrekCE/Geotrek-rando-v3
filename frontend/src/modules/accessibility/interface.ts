export interface RawAccessibility {
  id: string;
  name: string;
  pictogram: string;
}

export interface Accessibility {
  id: string;
  name: string;
  pictogramUri: string;
}
export interface AccessibilityLevel {
  id: number;
  name: { [language: string]: string };
}

export interface AccessibilityDictionnary {
  [id: string]: Accessibility;
}
