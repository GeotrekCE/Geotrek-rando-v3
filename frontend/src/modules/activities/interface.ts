export interface RawActivity {
  id: number;
  order: null;
  pictogram: string;
}

export interface RawUniqueActivity extends RawActivity {
  name: { [languageKey: string]: string | null };
}

export interface RawListActivity extends RawActivity {
  name: string;
}

export interface Activity {
  pictogram: string;
  name: string;
}

export interface ActivityChoices {
  [value: string]: Activity;
}

export interface ActivityFilter extends Activity {
  id: string;
  type: 'PRACTICE' | 'OUTDOOR_PRACTICE' | 'CATEGORY' | 'TOURISTIC_EVENT_TYPE';
}
