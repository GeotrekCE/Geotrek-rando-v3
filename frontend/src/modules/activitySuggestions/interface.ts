import { OutdoorSite } from 'modules/outdoorSite/interface';
import { TrekResult } from 'modules/results/interface';
import { TouristicContentDetails } from 'modules/touristicContent/interface';
import { TouristicEvent } from 'modules/touristicEvent/interface';

export interface ActivitySuggestion {
  type: 'events' | 'trek' | 'service' | 'outdoor';
  results: (TouristicContentDetails[] | TrekResult[] | OutdoorSite[] | TouristicEvent[])[];
  titleTranslationId: string;
  iconUrl: string;
  ids: string[];
}
