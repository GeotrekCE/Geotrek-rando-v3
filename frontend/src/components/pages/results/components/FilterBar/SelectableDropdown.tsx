/* eslint-disable @typescript-eslint/no-unsafe-return */

/**
 * We disable unsafe return beccause the lib react-select types it styles with any
 */

import { ReactElement } from 'react';
import { colorPalette } from 'stylesheet';
import Select from 'react-select';

interface Props {
  name: string;
  options: {
    value: string;
    label: string;
  }[];
  placeholder: string;
}

const colourStyles = {
  control: (styles: any) => ({
    ...styles,
    backgroundColor: 'white',
    boxShadow: 'none',
    minWidth: '150px',
    borderColor: colorPalette.filter.borderColor,
    ':hover': {
      borderColor: colorPalette.filter.borderColor,
    },
  }),
  option: (styles: any) => {
    return {
      ...styles,
      backgroundColor: colorPalette.filter.background,
      color: colorPalette.filter.color,
      ':hover': {
        backgroundColor: colorPalette.filter.hover.background,
        color: colorPalette.filter.hover.color,
      },
    };
  },
  multiValue: (styles: any) => {
    return {
      ...styles,
      backgroundColor: colorPalette.filter.selected.background,
    };
  },
  input: (styles: any) => ({ ...styles, backgroundColor: 'black' }),
  placeholder: (styles: any) => ({ ...styles, color: colorPalette.filter.placeholder.color }),
};

export const SelectableDropdown = (props: Props): ReactElement => (
  <Select
    options={props.options}
    isClearable={false}
    isSearchable={false}
    name={props.name}
    placeholder={props.placeholder}
    isMulti
    classNamePrefix="select"
    closeMenuOnSelect={false}
    styles={colourStyles}
  />
);
