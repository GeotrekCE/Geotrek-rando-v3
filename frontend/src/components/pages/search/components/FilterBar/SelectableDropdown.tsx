/* eslint-disable @typescript-eslint/no-unsafe-return */

/**
 * We disable unsafe return because the lib react-select types it styles with any
 */

import { ReactElement } from 'react';
import Select, { ValueType } from 'react-select';
import { Option } from 'modules/filters/interface';
import { useIntl } from 'react-intl';
import { colorPalette, sizes } from 'stylesheet';

interface Props {
  name: string;
  options: Option[];
  placeholder: string;
  setFilterSelectedOptions: (options: Option[]) => void;
  selectedFilters: Option[];
  filterType: 'SINGLE' | 'MULTIPLE';
  closeMenuOnSelect?: boolean;
}

const colourStyles = {
  control: (styles: any) => ({
    ...styles,
    backgroundColor: 'white',
    boxShadow: 'none',
    minWidth: '160px',
    borderColor: colorPalette.filter.borderColor,
    ':hover': {
      borderColor: colorPalette.filter.borderColor,
    },
    minHeight: sizes.button,
  }),
  option: (styles: any, { data }: { data: Option }) => {
    return {
      ...styles,
      backgroundColor: colorPalette.filter.background,
      color: colorPalette.filter.color,
      ':hover': {
        backgroundColor: colorPalette.filter.hover.background,
        color: colorPalette.filter.hover.color,
      },
      display: 'flex',
      alignItems: 'center',
      ':before': {
        content: data.pictogramUrl !== undefined ? '" "' : undefined,
        background: data.pictogramUrl !== undefined ? `url(${data.pictogramUrl})` : '',
        display: 'block',
        marginRight: 8,
        marginLeft: 4,
        width: 24,
        height: 24,
        backgroundSize: 'contain',
        backgroundRepeat: 'no-repeat',
        backgroundPosition: 'center',
        borderRadius: '50%',
        padding: 4,
        backgroundColor: colorPalette.primary1,
        backgroundOrigin: 'content-box',
      },
    };
  },
  multiValue: (styles: any, { data }: { data: Option }) => {
    return {
      ...styles,
      padding: '4px 0',
      backgroundColor: colorPalette.filter.selected.background,
      ':before': {
        content: data.pictogramUrl !== undefined ? '" "' : undefined,
        background: data.pictogramUrl !== undefined ? `url(${data.pictogramUrl})` : '',
        display: 'block',
        marginRight: 2,
        marginLeft: 4,
        width: 24,
        backgroundSize: 'contain',
        backgroundRepeat: 'no-repeat',
        backgroundPosition: 'center',
        borderRadius: '50%',
        padding: 4,
        backgroundColor: colorPalette.primary1,
        backgroundOrigin: 'content-box',
      },
    };
  },
  input: (styles: any) => ({ ...styles, backgroundColor: 'black' }),
  placeholder: (styles: any) => ({ ...styles, color: colorPalette.filter.placeholder.color }),
};

const computeAction = (action: ValueType<Option, true>): Option[] => {
  if (action === undefined || action === null) return [];
  if (action.length >= 0) return [...action];
  //@ts-ignore We ignore because the issue is between a readonly array and an array
  return [action];
};

export const SelectableDropdown = (props: Props): ReactElement => {
  const intl = useIntl();
  return (
    <Select
      closeMenuOnSelect={false}
      {...props}
      isClearable={props.filterType === 'SINGLE'}
      isSearchable={false}
      placeholder={intl.formatMessage({ id: props.placeholder })}
      classNamePrefix="select"
      isMulti={props.filterType === 'MULTIPLE' ? true : undefined}
      styles={colourStyles}
      value={props.selectedFilters}
      onChange={(action: ValueType<Option, true>) => {
        const options = computeAction(action);
        props.setFilterSelectedOptions(options);
      }}
    />
  );
};
