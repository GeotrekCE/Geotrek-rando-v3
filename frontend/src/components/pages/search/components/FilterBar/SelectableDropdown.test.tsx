import { Option } from 'modules/filters/interface';
import { render } from 'services/testing/reactTestingLibraryWrapper';
import { SelectableDropdown } from './SelectableDropdown';

test('SelectableDropdown shoudl render properly', () => {
  const component = render(
    <SelectableDropdown
      name="activities"
      placeholder="search.filters.DIFFICULTY"
      options={[
        {
          label: 'Très Facile',
          value: '1',
        },
        {
          label: 'Facile',
          value: '2',
        },
      ]}
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      setFilterSelectedOptions={(values: Option[]) => {}}
      selectedFilters={[
        {
          label: 'Très Facile',
          value: '1',
        },
      ]}
    />,
  );
  expect(component).toMatchSnapshot();
});
