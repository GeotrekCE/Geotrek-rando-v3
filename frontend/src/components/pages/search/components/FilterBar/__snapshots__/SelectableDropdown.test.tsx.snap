// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectableDropdown should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" css-2b097c-container"
      >
        <div
          class="select__control css-io5zd9-control"
        >
          <div
            class="select__value-container select__value-container--is-multi select__value-container--has-value css-g1d714-ValueContainer"
          >
            <div
              class="css-ya8gvn-multiValue select__multi-value"
            >
              <div
                class="css-12jo7m5 select__multi-value__label"
              >
                Très Facile
              </div>
              <div
                class="css-xb97g8 select__multi-value__remove"
              >
                <svg
                  aria-hidden="true"
                  class="css-6q0nyr-Svg"
                  focusable="false"
                  height="14"
                  viewBox="0 0 20 20"
                  width="14"
                >
                  <path
                    d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                  />
                </svg>
              </div>
            </div>
            <input
              aria-autocomplete="list"
              class="css-62g3xt-dummyInput"
              id="react-select-2-input"
              readonly=""
              tabindex="0"
              value=""
            />
          </div>
          <div
            class="select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class="select__indicator-separator css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="select__indicator select__dropdown-indicator css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-6q0nyr-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div>
          <input
            name="activities"
            type="hidden"
            value="1"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" css-2b097c-container"
    >
      <div
        class="select__control css-io5zd9-control"
      >
        <div
          class="select__value-container select__value-container--is-multi select__value-container--has-value css-g1d714-ValueContainer"
        >
          <div
            class="css-ya8gvn-multiValue select__multi-value"
          >
            <div
              class="css-12jo7m5 select__multi-value__label"
            >
              Très Facile
            </div>
            <div
              class="css-xb97g8 select__multi-value__remove"
            >
              <svg
                aria-hidden="true"
                class="css-6q0nyr-Svg"
                focusable="false"
                height="14"
                viewBox="0 0 20 20"
                width="14"
              >
                <path
                  d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                />
              </svg>
            </div>
          </div>
          <input
            aria-autocomplete="list"
            class="css-62g3xt-dummyInput"
            id="react-select-2-input"
            readonly=""
            tabindex="0"
            value=""
          />
        </div>
        <div
          class="select__indicators css-1hb7zxy-IndicatorsContainer"
        >
          <span
            class="select__indicator-separator css-1okebmr-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            class="select__indicator select__dropdown-indicator css-tlfecz-indicatorContainer"
          >
            <svg
              aria-hidden="true"
              class="css-6q0nyr-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div>
        <input
          name="activities"
          type="hidden"
          value="1"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
